### PR TITLE
Trap a permissions error on redcap instrument

### LINKED ISSUE #343: Error in one instrument in the manager dashboard stalls all processing
Nothing ran on a project with a bunch of assessors launched (Neg_Val_SCZ starting mid-Dec). The only problem seemed to be a typo in the inputs.xnat.etc variable name in a single one of the instruments. Can dax run the ones that are ok and skip the ones that fail?

### PR DESCRIPTION
**Pull request by baxpr**
fixes #343 probably

### CODE REVIEW

**Comment by baxpr** [comment]
Oops it doesn't resolve that issue. This was a different problem.

**Comment by baxpr** [comment]
Added a work-around for when proctype is missing. That should get us going until the FS proctype issue is sorted out (http://xnatbug.vandyxnat.org/bugzilla/show_bug.cgi?id=2393)

**Comment by baxpr** [comment]
Found the proctype issue in another place too. Now tested - logs an error and proceeds when the missing proctype is encountered.

**Comment by baxpr** [comment]
Last update was to continue to next assessor instead of return from the entire function when an assessor with no inputs is found. I think that would previously have been skipping some assessors?

File: dax/processor_parser.py [hunk 1]
```diff
@@ -942,16 +946,24 @@ def compare_to_existing(csesses, proc_type, parameter_matrix):
...
+                    LOGGER.warn(f'Empty inputs - skipping {casr.label()}')
+                    continue
 
             for pi, p in enumerate(parameter_matrix):
                 if inputs == p:
-                    # BDB 6/5/21 do we ever have more than one assessor
-                    # with the same set of inputs?
+                    # BDB  6/5/21 do we ever have more than one assessor
+                    #             with the same set of inputs?
+                    # BPR 2/28/22 in principle we could
```
Comment (bud42) [inline dax/processor_parser.py:966]: how?

Comment (baxpr) [reply dax/processor_parser.py:966]: Isn't this actually common? Two different T1 segmentation pipelines that only take the T1 as input, for example.

Comment (baxpr) [reply dax/processor_parser.py:966]: If I am not understanding something, I may well have messed up that bit of the code so....

Comment (bud42) [reply dax/processor_parser.py:966]: But of the same proc type?

Comment (baxpr) [reply dax/processor_parser.py]: I did mess it up - the pi,p loop should be inside the if proctype_match. New commit

Comment (baxpr) [reply dax/processor_parser.py]: ... or should it. Argh I don't understand the logic here

Comment (baxpr) [reply dax/processor_parser.py]: I think it should. @bud42 can you verify the logic now is the same as it was before, just with added error catching?

**Review by bud42 — APPROVED** [review]

**Pull request merged**